ext/soap/soap.c

```diff
@@ -673,20 +683,22 @@ PHP_METHOD(SoapFault, __toString)
 /* {{{ SoapVar constructor */
 PHP_METHOD(SoapVar, __construct)
 {
-	zval *data, *type, *this_ptr;
+	zval *data, *this_ptr;
+	zend_long type;
+	zend_bool type_is_null = 1;
 	char *stype = NULL, *ns = NULL, *name = NULL, *namens = NULL;
 	size_t stype_len = 0, ns_len = 0, name_len = 0, namens_len = 0;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "z!z|ssss", &data, &type, &stype, &stype_len, &ns, &ns_len, &name, &name_len, &namens, &namens_len) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "z!l!|ssss", &data, &type, &type_is_null, &stype, &stype_len, &ns, &ns_len, &name, &name_len, &namens, &namens_len) == FAILURE) {
 		RETURN_THROWS();
 	}
 
 	this_ptr = ZEND_THIS;
-	if (Z_TYPE_P(type) == IS_NULL) {
+	if (type_is_null) {
 		add_property_long(this_ptr, "enc_type", UNKNOWN_TYPE);
 	} else {
-		if (zend_hash_index_exists(&SOAP_GLOBAL(defEncIndex), Z_LVAL_P(type))) {
-			add_property_long(this_ptr, "enc_type", Z_LVAL_P(type));
```

Comment:
Huh. Looks like this fixes a potential crash when type is non-null but also non-int.